### PR TITLE
Download SP logs

### PR DESCRIPTION
**Pull request by rakanalh**
This script does the following:
1. Figure out the run number of each of the existing scenarios
2. Download the logs for each of the nodes involved in the latest scenario run
3. Looks through the logs and searches for errors / exceptions and reports them

## Description

Fixes: #5636

Please, describe what this PR does in detail:
- If it's a bug fix, detail the root cause of the bug and how this PR fixes it.
- If it's a new feature, describe the feature this PR is introducing and design decisions.
- If it's a refactoring, describe why it is necessary. What are its pros and cons in respect to the previous code, and other possible design choices.

In the Raiden root directory, to invoke:
```
 ./tools/debugging/extract_sp_exceptions.sh ./sp-nightly-dir # or whatever target directory you like
```

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: tools/debugging/extract_sp_exceptions.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+### STYLES: IGNORE
+RESET="\e[0m"
+BOLD="\e[1m"
+###
+
+
+function display_usage {
+    echo "This script does the following:"
+    echo "1. Figure out the run number of each of the existing scenarios"
```
Comment (LefterisJP) [inline tools/debugging/extract_sp_exceptions.sh]: super nitcpick: 
1. Figure -> Figures
2. Download -> Downloads

File: tools/debugging/extract_sp_exceptions.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    echo "2. Download the logs for each of the nodes involved in the latest scenario run"
+    echo "3. Looks through the logs and searches for errors / exceptions and reports them"
+    echo -e "\n\nUSAGE: extrach_sp_exceptions.sh [DESTINATION_DIR]\n"
```
Comment (LefterisJP) [inline tools/debugging/extract_sp_exceptions.sh]: exctrach -> extract

File: tools/debugging/extract_sp_exceptions.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    echo -e "Where\n"
+    echo -e "\tDESTINATION_URL is optional and is used to specify the path prefix for the downloaded files. Otherwise the current directory is used\n"
+}
+
+function print_bold {
+    echo -e "${BOLD}$1${RESET}"
+}
+
+# if less than two arguments supplied, display usage
```
Comment (LefterisJP) [inline tools/debugging/extract_sp_exceptions.sh]: The comment does not match the if conditional. Change one of them

File: tools/debugging/extract_sp_exceptions.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+if [[ $1 != "" ]]; then
+    DESTINATION_DIR=$1
+else
+    DESTINATION_DIR=$(pwd)
+fi
+
+print_bold "Downloading PFS logs"
+
+ssh root@services-dev.raiden.network 'cd raiden-services/deployment/; docker-compose logs pfs-goerli | gzip' > ${DESTINATION_DIR}/pfs.log.gz
+
```
Comment (LefterisJP) [inline tools/debugging/extract_sp_exceptions.sh]: You can check here for ssh command failure and stop the script. Something like

```sh
if [[ $? -ne 0 ]]; then
    echo "Error: ssh to server failed"
    exit 1
fi
```

Same for other commands. But may be an overkill.

**Comment by agatsoh** [comment]
I have curl version 7.58.0. So I do not have the `--no-progress-meter` option. Is it possible to use some other options on the curl command ?  I am on linux mint Tricia and updating upgrading uninstalling  installing gives the same version, I might need to install higher version from source or something. I am prepared for either option. So @rakanalh please let me know if its possible ? Thanks a lot !!!

File: tools/debugging/extract_sp_exceptions.sh (hunk 4 above)
Comment (rakanalh) [reply tools/debugging/extract_sp_exceptions.sh]: Added

**Comment by rakanalh** [comment]
Addressed @agatsoh's comment about `--no-progress-meter` by replacing it with `--silent` here: https://github.com/raiden-network/raiden/pull/5643/commits/a207ed787da5eba2323799972cddb170064daa37

**Pull request merged**